packages/protocol/test/L1/TaikoL1.test.ts

```diff
@@ -85,6 +87,435 @@ describe("TaikoL1", function () {
             expect(provers).to.be.empty;
         });
     });
+
+    describe("halt()", async function () {
+        it("should revert called by nonOwner", async function () {
+            const initiallyHalted = await taikoL1.isHalted();
+            expect(initiallyHalted).to.be.eq(false);
+            const signers = await ethers.getSigners();
+            await expect(
+                taikoL1.connect(signers[1]).halt(true)
+            ).to.be.revertedWith("Ownable: caller is not the owner");
+
+            const isHalted = await taikoL1.isHalted();
+            expect(isHalted).to.be.eq(false);
+        });
+
+        it("should not revert when called by owner", async function () {
+            const initiallyHalted = await taikoL1.isHalted();
+            expect(initiallyHalted).to.be.eq(false);
+            await taikoL1.halt(true);
+            const isHalted = await taikoL1.isHalted();
+            expect(isHalted).to.be.eq(true);
+        });
+    });
+
+    describe("proposeBlock()", async function () {
+        it("should revert when size of inputs is les than 2", async function () {
+            await expect(
+                taikoL1.proposeBlock([randomBytes32()])
+            ).to.be.revertedWith("L1:inputs:size");
+        });
+
+        it("should revert when halted", async function () {
+            await taikoL1.halt(true);
+            await expect(
+                taikoL1.proposeBlock([randomBytes32()])
+            ).to.be.revertedWith("0x1");
+        });
+    });
+
+    describe("commitBlock()", async function () {
+        it("should revert when size of inputs is les than 2", async function () {
+            await expect(
+                taikoL1.proposeBlock([randomBytes32()])
+            ).to.be.revertedWith("L1:inputs:size");
+        });
+
+        it("should revert when halted", async function () {
+            await taikoL1.halt(true);
+            await expect(
+                taikoL1.proposeBlock([randomBytes32()])
+            ).to.be.revertedWith("0x1");
+        });
+    });
+
+    describe("whitelisting()", async function () {
```

Comment:
seems `whitelisting` feature will be removed after tokenomics get merged? ref: https://github.com/taikoxyz/taiko-mono/pull/2096